### PR TITLE
Add Most Read container to WS AMP (STY & article)

### LINKED ISSUE #10012: Add Most read to WS articles pages (Test environment only)
**Is your feature request related to a problem? Please describe.**
Add most read to amp article pages
Most read can be seen on canonical pages
https://www.bbc.com/kyrgyz/articles/c72x8jgx17do
but is missing from the AMP version
https://www.bbc.com/kyrgyz/articles/c72x8jgx17do.amp

The configuration for most read  in the Simorgh config
https://github.com/bbc/simorgh/tree/latest/src/app/lib/config/services
```
mostRead: {
      header: 'De one we dem de read well well',
      lastUpdated: 'De one we dem update for:',
      numberOfItems: 10,
      hasMostRead: true,
    },

```
**Describe the solution you'd like**
Add most read to the AMP article pages.

**Describe alternatives you've considered**

**Fallback**
Don't show component if data fails to load. Might need to show placeholder instead because of the responsive layout. Could use `amp-bind` to hide the title if data does not load

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Checklist**
- [ ] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**

### PR DESCRIPTION
**Pull request by MuhammadAHussain**
Resolves https://github.com/bbc/simorgh/issues/10012

**Overall change:**
This will add Most Read to STY and articles page types on amp. It will not render on other pages types such as front page.

**Code changes:**

- Create new amp most read component.
- Add cypress tests which test most read for article and STY pages.
- Create generateCSPHash function in utilities.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project
- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: cypress/integration/pages/articles/testsForAMPOnly.js [hunk 1]
```diff
@@ -60,6 +63,98 @@ export const testsThatFollowSmokeTestConfigForAMPOnly = ({
...
+            const mostReadRecords = mostReadJson.totalRecords;
+            cy.fixture(`toggles/${config[service].name}.json`).then(toggles => {
+              const mostReadIsEnabled = path(['mostRead', 'enabled'], toggles);
+              cy.log(
+                `Most read container toggle enabled? ${mostReadIsEnabled}`,
+              );
+              if (
+                Cypress.env('APP_ENV') !== 'live' &&
+                mostReadIsEnabled &&
+                mostReadRecords > 5
```
Comment (LilyL0u) [inline cypress/integration/pages/articles/testsForAMPOnly.js]: Should this be >= 5 (or > 4) as we want to run the test if there are 5 records? I think 5 is the minimum.

File: cypress/integration/pages/articles/testsForAMPOnly.js [hunk 2]
```diff
@@ -60,6 +63,98 @@ export const testsThatFollowSmokeTestConfigForAMPOnly = ({
...
```
Comment (LilyL0u) [inline cypress/integration/pages/articles/testsForAMPOnly.js]: same as above

File: cypress/integration/pages/articles/testsForAMPOnly.js [hunk 3]
```diff
@@ -60,6 +63,98 @@ export const testsThatFollowSmokeTestConfigForAMPOnly = ({
...
```
Comment (LilyL0u) [inline cypress/integration/pages/articles/testsForAMPOnly.js]: same

**Comment by LilyL0u** [comment]
Tests look good :) I just added a comment for a small change in each one that we missed when discussing the minimum number of records for running the test. I just added it on one test file, but it is the same for both.

File: cypress/integration/pages/articles/testsForAMPOnly.js (hunk 1 above)
Comment (MuhammadAHussain) [reply cypress/integration/pages/articles/testsForAMPOnly.js]: Done 👍

File: src/app/containers/MostRead/Amp/index.jsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+          {AMP_LIST_JS}
+          {AMP_MUSTACHE_JS}
+          {AMP_SCRIPT_JS}
+          <meta
+            name="amp-script-src"
+            content={generateCSPHash(
+              onlyinnerscript,
+              'sha384',
+              'utf8',
+              'base64',
```
Comment (amoore108) [inline src/app/containers/MostRead/Amp/index.jsx:87]: Looks great, excellent work @MuhammadAHussain @DarioR01 🥳 

Keen to know the performance impact (if any) on running the hash generation each time. Probably minimal, just curious more than anything.

**Review by amoore108 — APPROVED** [review]

**Review by mfonofm — APPROVED** [review]

File: src/app/containers/MostRead/Amp/index.jsx (hunk 4 above)
Comment (MuhammadAHussain) [reply src/app/containers/MostRead/Amp/index.jsx:87]: Yh we were curiously about that too but not too sure how to measure it.

**Review by Louis-Matsika — APPROVED** [review]

**Pull request merged**